src/ConfigurationHelper.php

```diff
@@ -0,0 +1,539 @@
+      if (array_key_exists(Constants::FACET_SELECTION_FIELD, $entity_fields)) {
+        $this->indexAddFacetField($index);
+      }
+      if (array_key_exists(Constants::TITLE_SORT_FIELD, $entity_fields)) {
+        $this->indexAddTitleSortField($index);
+      }
+      $index->save();
+    }
+    if ($view = $this->getView()) {
+      $this->viewSetViewMode($view, $entity_type_id, $entity_bundle);
+      $view->save();
+    }
+    $this->blockAddContentType(Constants::CHANNEL_SEARCH_BLOCK, $entity_bundle);
+  }
+
+  /**
+   * Act on Directory channel field being removed.
+   */
+  public function deletedDirectoryChannelField(FieldConfigInterface $field): void {
+    // Only working for nodes at the moment.
+    $entity_type_id = $field->getTargetEntityTypeId();
+    $entity_bundle = $field->getTargetBundle();
+    // Index changes.
+    if ($index = $this->getIndex()) {
+      $this->indexRemoveBundle($index, $entity_type_id, $entity_bundle);
+    }
+    $this->blockRemoveContentType(Constants::CHANNEL_SEARCH_BLOCK, $entity_bundle);
+  }
+
+  /**
+   * Act on Directory facet field being added.
+   */
+  public function insertedFacetField(FieldConfigInterface $field): void {
+    if ($index = $this->getIndex()) {
+      $this->indexAddFacetField($index);
+      $index->save();
+    }
+    $this->createFacet();
+  }
+
+  /**
+   * Act on Directory title sort field being added.
+   */
+  public function insertedTitleSortField(FieldConfigInterface $field): void {
+    if ($index = $this->getIndex()) {
+      $this->indexAddTitleSortField($index);
+      $index->save();
+    }
+  }
+
+  /**
+   * Create new config entity from given config file.
+   *
+   * @param string $entity_type
+   *   Example: facets_facet.
+   * @param string $config_path
+   *   Example: modules/foo/config/bar.
+   * @param string $config_filename
+   *   Example: views.view.bar.
+   */
+  public function importConfigEntity(string $entity_type, string $config_path, string $config_filename): bool {
+    $config_src = new ConfigFileStorage($config_path);
+    if (empty($config_src)) {
+      return FALSE;
+    }
+
+    $config_values = $config_src->read($config_filename);
+    if (empty($config_values)) {
+      return FALSE;
+    }
+
+    try {
+      $this->entityTypeManager->getStorage($entity_type)->create($config_values)->save();
+    }
+    catch (\Exception $e) {
+      $this->logger->error('Failed to create new config entity: %filename.  Error: %msg', [
+        '%filename' => $config_filename,
+        '%msg' => $e->getMessage(),
+      ]);
+
+      return FALSE;
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Setup indexing on the Facet selection field of Directory entries.
+   *
+   * This assumes that the localgov_directory_facets_select field is part of a
+   * Directory entry content type.
+   *
+   * @param \Drupal\search_api\IndexInterface $index
+   *   The index to the facet field to.
```

Comment:
Sorry, I didn't get this sentence in line 246.  Did you mean to say "The index to add the facet field to"?

Reply:
Corrected in https://github.com/localgovdrupal/localgov_directories/pull/233/commits/b187781ad0067d98be3ee637f3ef155c6c967ba2